src/rpc/misc.cpp

```diff
@@ -1258,6 +1259,150 @@ UniValue getaddressbalance(const JSONRPCRequest& request)
     return result;
 }
 
+UniValue RanksToUniValue(CAmount lottery_anv, const Ranks& ranks, size_t total) {
+
+    UniValue rankarr(UniValue::VARR);
+    for(const auto& r : ranks) {
+        UniValue o(UniValue::VOBJ);
+
+        //percentile to two digits
+        double percentile = 
+            std::floor((static_cast<double>(r.second) / static_cast<double>(total)) * 10000.0) / 100.0;
+
+        o.push_back(Pair("rank", total - r.second));
+        o.push_back(Pair("percentile", percentile));
+        o.push_back(Pair("anv", r.first.anv));
+
+        double anv_percent = 
+            (static_cast<double>(r.first.anv) / static_cast<double>(lottery_anv));
+
+        o.push_back(Pair("anvpercent", anv_percent));
+        rankarr.push_back(o);
+    }
+    return rankarr;
+}
+
+UniValue getaddressrank(const JSONRPCRequest& request)
```

Comment:
Some DRY + duplication between this and `getaddressleaderboard`. I don't have a concrete suggestion for how to address it but something to consider.